### PR TITLE
Pin docker compose to 2.29.7 to avoid regression bug

### PR DESCRIPTION
**Pull request by klutchell**
In v2.32.1 there is an issue with depends_on not being
propagated when two compose files are merged via the config command
and one extends the other.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    -   id: shellcheck
+
+-   repo: https://github.com/rhysd/actionlint
+    rev: v1.7.3
+    hooks:
+    -   id: actionlint-docker
+        args: ["-color", "-ignore", ":info:", "-ignore", ":style:"]
+
+-   repo: https://github.com/igorshubovych/markdownlint-cli
+    rev: v0.42.0
```
Comment (joshbwlng) [inline .pre-commit-config.yaml:29]: I see there's a v0.43.0 available, any reason to not use it?

Comment (klutchell) [reply .pre-commit-config.yaml:29]: No reason, just copy-paste. I've updated it now @joshbwlng

Comment (joshbwlng) [reply .pre-commit-config.yaml]: @klutchell What about the others?

Comment (klutchell) [reply .pre-commit-config.yaml]: Good shout, I've moved the pre-commit changes to [another PR](https://github.com/product-os/self-hosted-runners/pull/420) where I will add support for Renovate to bump the versions. It keeps this PR simple to un-break the runners.

**Review by aethernet — APPROVED** [review]

**Pull request merged**